### PR TITLE
Add PEP 484 type hints.

### PR DESCRIPTION
**Pull request by pcorpet**
See https://www.python.org/dev/peps/pep-0484/
For #94

This is not complete, but just some hints that are already valid but on which we can iterate in further commits.

### CODE REVIEW

File: requests_mock/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Stubs for requests_mock
+
+from requests_mock.adapter import ANY as ANY, Adapter as Adapter
+from requests_mock.exceptions import MockException as MockException, NoMockAddress as NoMockAddress
+from requests_mock.mocker import DELETE as DELETE, GET as GET, HEAD as HEAD, Mocker as Mocker, MockerCore as MockerCore, OPTIONS as OPTIONS, PATCH as PATCH, POST as POST, PUT as PUT, mock as mock
+from requests_mock.request import _RequestObjectProxy as _RequestObjectProxy
+from requests_mock.response import CookieJar as CookieJar, create_response as create_response
```
Comment (palfrey) [inline requests_mock/__init__.pyi:7]: Why all the `as` items for things that have the same name?

Comment (pcorpet) [reply requests_mock/__init__.pyi:7]: See the note on stubs in https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-no-implicit-reexport

**Comment by jamielennox** [comment]
Hey, just wanted to say I've seen this PR and appreciate the work. I've got some family stuff going on at the moment (good things) that mean I've gotten as far as opening it in pycharm to realise it doesn't pick them up or of the box - but not even looked at why.

As far as I'm concerned anything here is better than nothing so if you two (@palfrey @pcorpet) agree that these are working and a good starting point I'll merge it and we can improve it incrementally from there.

Just FYI, my intent with dropping py2 had been to wait until request V3 was released, figure out the necessary changes to do a new major version of requests-mock and have that be py3 only. Requests3 appears to be a bit of a drama so it might take longer than I expected.

**Comment by palfrey** [comment]
I'm happy with the work here as it removed any need I had for maintaining my own mocks in my project

**Comment by pcorpet** [comment]
I confirm I think this is a good starting point. Congrats @jamielennox for your familly stuff and thanks for answering the PR while you're busy.

**Review by jamielennox — APPROVED** [review]

**Comment by intgr** [comment]
Thanks for this, it is very useful for my project :)

**Pull request merged**